FuSa-addon/scripts/REQIF_export.groovy

```diff
@@ -51,33 +44,46 @@ def traverseTree(thisNode, parent_identifier)
     }
 
     thisNode.children.each{
-        traverseTree(it, identifier)
+        traverseTree(it, identifier, requirementlists, relationlist)
     }
 }
 
-def makeHeader(XML, DATE)
+def makeHeader(XML)
 {
     XML.'THE-HEADER'{
-        XML.'REQ-IF-HEADER'('IDENTIFIER':'_adkamo'){
-            XML.'CREATION-TIME'(DATE)
-            XML.'REPOSITORY-ID'('?')
-            XML.'REQ-IF-TOOL-ID'('Freemind FuSa Extension v0.5')
-            XML.'REQ-IF-VERSION'('1.0')
-            XML.'SOURCE-TOOL-ID'('I guess this is only relevant for a round trip')
+        XML.'REQ-IF-HEADER'('IDENTIFIER':"_" + UUID.randomUUID().toString()){
+            XML.'CREATION-TIME'("${format.format(new Date())}")
+            XML.'REPOSITORY-ID'("__73f2bb4d-760e-44d9-99d3-5807d44b4928")
+            XML.'REQ-IF-TOOL-ID'("_5021db30-dd57-4612-bb75-98c4801619c0")
+            XML.'REQ-IF-VERSION'('1.1')
+            XML.'SOURCE-TOOL-ID'("_63165248-e740-4271-a5e1-03ddc648d5eb")
```

Comment:
Same applies as above, but If I read the spec right, this field is for roundtrip exchange of requirements going back and forth between tools, so for a first time export, I think it should be empty, not sure though.

Reply:
Lack of consideration.. sorry.
I will change header such as bellow.

XML.'REQ-IF-TOOL-ID'('Freemind FuSa Extension v0.5')
XML.'REQ-IF-VERSION'('1.0')
XML.'SOURCE-TOOL-ID'('Freeplane')


*Doors NG 7.0.2 exported header sample ( round trip mode )
Doors NG has reqif export function, so REQ-IF-TOOL-ID and SOURCE-TOOL-ID is same.

<THE-HEADER>
		<REQ-IF-HEADER IDENTIFIER="_5fef5cd0-1913-4611-8324-dd3074d54e65">
			<CREATION-TIME>2021-06-20T22:08:31.230Z</CREATION-TIME>
			<REPOSITORY-ID></REPOSITORY-ID>
			<REQ-IF-TOOL-ID>IBM Engineering Requirements 7.0.2</REQ-IF-TOOL-ID>
			<REQ-IF-VERSION>1.0</REQ-IF-VERSION>
			<SOURCE-TOOL-ID>IBM Engineering Requirements 7.0.2</SOURCE-TOOL-ID>
			<TITLE>IBM Engineering Requirements Sun Jun 20 22:08:31 EDT 2021</TITLE>
		</REQ-IF-HEADER>
</THE-HEADER>

*Jama exported header (export only mode, no round trip.)
Jama  Connect doesn't have reqif export function itself.
To export a reqif file, an user has to use External tool called Data Exchange.
REQ-IF-TOOL-ID is  Data Exchange and SOURCE-TOOL-ID is Jama Connect.
        <REQ-IF-HEADER IDENTIFIER="_7067c9b1-64e8-45da-9765-1af1dfae7185">
            <CREATION-TIME>2021-06-04T18:53:30.991+09:00</CREATION-TIME>
            <REQ-IF-TOOL-ID>Data Exchange 3.5.0</REQ-IF-TOOL-ID>
            <REQ-IF-VERSION>1.0</REQ-IF-VERSION>
            <SOURCE-TOOL-ID>Jama Connect</SOURCE-TOOL-ID>
            <TITLE>APCM</TITLE>
        </REQ-IF-HEADER>

Reply:
Hi Ryushima san,

thanks, looks good, still one more typo left though, it should be
`XML.'REQ-IF-TOOL-ID'('Freeplane FuSa Extension v0.5')`